BASE/Test/Microsoft.ApplicationInsights.Test/Microsoft.ApplicationInsights.Tests/Channel/TransmissionTest.cs

```diff
@@ -363,6 +366,87 @@ public async Task SendAsyncReturnsCorrectHttpResponseWrapperWithRetryHeaderWhenN
                 }
 
             }
+
+#if NETCOREAPP2_1
+            [TestMethod]
+            public async Task SendAsyncLogsIngestionReponseTimeEventCounter()
+            {
+                var handler = new HandlerForFakeHttpClient
+                {
+                    InnerHandler = new HttpClientHandler(),
+                    OnSendAsync = (req, cancellationToken) =>
+                    {
+                        return Task.FromResult<HttpResponseMessage>(new HttpResponseMessage());
+                    }
+                };
+
+                using (var fakeHttpClient = new HttpClient(handler))
+                {
+                    // Instantiate Transmission with the mock HttpClient
+                    Transmission transmission = new Transmission(testUri, new byte[] { 1, 2, 3, 4, 5 }, fakeHttpClient, string.Empty, string.Empty);
+
+                    using (var listener = new TestEventListener())
+                    {
+                        var eventCounterArguments = new Dictionary<string, string>
+                        {
+                            {"EventCounterIntervalSec", "1"}
+                        };
+
+                        listener.EnableEvents(CoreEventSource.Log, EventLevel.LogAlways, (EventKeywords)AllKeywords, eventCounterArguments);
+
+                        HttpWebResponseWrapper result = await transmission.SendAsync();
+                        //Sleep for a second as the event counter is sampled on a second basis - EventCounterIntervalSec
+                        Thread.Sleep(TimeSpan.FromSeconds(1));
+
+                        // VERIFY
+                        // We validate by checking SDK traces.
+                        var allTraces = listener.Messages.ToList();
+                        var traces = allTraces.Where(item => item.EventName == "EventCounters").ToList();
+                        Assert.AreEqual(1, traces.Count);
+                        var payload = (IDictionary<string, object>)traces[0].Payload[0];
+                        Assert.AreEqual("IngestionEndpoint-ResponseTimeMsec", payload["Name"].ToString());
```

Comment:
validate the payload value of actual responsetime. by introducing "fake delay" in httpclient, this should be equal or more than the delay in httpclient.

Reply:
Done